README.md

```diff
@@ -379,3 +259,53 @@ Make sure:
 1. Add pass at the end of the function.
 1. Don’t assume the default setting will always work. Call out the expected settings in the prerequisite and they should be set explicitly at the beginning of the test. Use the fixture client or others to reset any changed settings back to default.
 
+### Test Fixture <a name="add_new_fixture"/>
+- Shareable fixture
+    - Scope must be `session` or `module`
+    - Able to be tear down gracefully
+    - Be placed in `fixtures/<topic_file>.py`
+- Specific fixture
+    - Be placed in test file
+
+### Add New Tests <a name="add_new_test" />
+
+Here are the general guidelines for adding a new e2e test:
+- File name should starts with `test_`
+- Add Markers for new Test
+    - Priority markers: `p0`, `p1` and `p2`
+    - At least add 1 topic marker
+- New e2e Test(s) should be corresponding to **ONLY** one [manual test case][Harvester manual test cases],
+    - When the Manual test case having several sub tests, use `TestClass` for it.
+- ONLY update old e2e test case when corresponding manual test case going to be updated
+- Any constants should be unique or inputted by configuration
+- Any delay (sleep) should have explicit reason
+- Make sure tear down is safe for other tests
+- Tests should be using [fixtures][pytest fixtures] as much as possible as the [fixtures][pytest fixtures] should be able to cleanup after itself upon exit. Otherwise, the test must cleanup whatever artifacts it create upon exit.
```

Comment:
Tests should be using [fixtures][pytest fixtures] as much as possible as the [fixtures][pytest fixtures] should be able to clean up after themselves upon exit. Otherwise, the test must clean up whatever artifacts it creates upon exit.